rfd/0080-piv-private-key.md

```diff
@@ -0,0 +1,392 @@
+
+This change will require changing the `require_session_mfa` fields above from a `bool` to a `string`. This will be handled by introducing a new proto field and custom marshalling logic to maintain interoperability between new and old servers and clients. See https://github.com/gravitational/teleport/pull/12054 for an example of this.
+
+#### Enforcement - Attestation
+
+In order to enforce user private key usage, we need to take a certificate's public key and tie it back to a trusted hardware device, which can be done with attestation, as explained [above](#attestation).
+
+Attestation will be handled during the normal login/certificate signing process by adding a new `AttestationRequest` field to login requests. For all login paths, we need to include the `AttestationRequest` field in the http request objects:
+
+```go
+// lib/client/weblogin.go
+type SSOLoginConsoleReq struct {
+  ...
+	AttestationRequest AttestationRequest `json:"attestation_request,omitempty"`
+}
+
+type CreateSSHCertReq struct {
+  ...
+	AttestationRequest AttestationRequest `json:"attestation_request,omitempty"`
+}
+
+type AuthenticateSSHUserRequest struct {
+  ...
+	AttestationRequest AttestationRequest `json:"attestation_request,omitempty"`
+}
+```
+
+For SSO login, the `AttestationRequest` field also needs to be added to each SSO auth request type (`OIDCAuthRequest`, `SAMLAuthRequest`, `GithubAuthRequest`), so we will make `AttestationRequest` a proto type.
+
+```proto
+// AttestationRequest is a request to attest an attestation request.
+message AttestationRequest {
+  oneof attestation_request {
+    // yubikey_attestation_request is a request to attest a specific YubiKey PIV slot.
+    YubiKeyAttestationRequest yubikey_attestation_request = 1;
+  }
+}
+
+// YubiKeyAttestationRequest is a request to attest a specific YubiKey PIV slot.
+// See https://pkg.go.dev/github.com/go-piv/piv-go@v1.9.0/piv#Attestation
+message YubiKeyAttestationRequest {
+  // slot_cert is an attestation certificate generated from a YubiKey PIV
+  // slot's public key and signed by the YubiKey's attestation certificate.
+  bytes slot_cert = 1;
+
+  // attestation_cert is a YubiKey's unique attestation certificate signed by a Yubico CA.
+  bytes attestation_cert = 2;
+}
+```
+
+When the Auth Server receives a login request, it will check the attached attestation request, and verify it. In addition to verifying the certificate chain for the hardware private key, the resulting attestation object will provide information about the private key, including;
+ - Device information including serial number, model, and version 
+ - Configured Touch (And PIN) Policies
+
+Currently, we are only interested in verifying the certificate chain for the public key, and checking its private key policy, so the resulting attestation response will look like:
```

Comment:
Who is attestation response meant for? The client?

If yes, then why is the public key contained within it?